DEVELOPER_GUIDE.md

```diff
@@ -292,3 +302,28 @@ original PR with an appropriate label `backport <backport-branch-name>` is merge
 run successfully on the PR. For example, if a PR on main needs to be backported to `2.x` branch, add a label
 `backport 2.x` to the PR and make sure the backport workflow runs on the PR along with other checks. Once this PR is
 merged to main, the workflow will create a backport PR to the `2.x` branch.
+
+## Building on Lucene Version updates
+There may be a Lucene version update that can affect your workflow causing errors like
+`java.lang.NoClassDefFoundError: org/apache/lucene/codecs/lucene99/Lucene99Codec`.
+This results in having issues with not being able to do `./gradlew run` or `./gradlew build`.
+
+You can check this [PR](https://github.com/opensearch-project/k-NN/pull/2195) as an example of this event happening
+
+**Follow the steps to remedy the gradle run issue.**
```

Comment:
We need to include step for checking if the real problem has been fixed in neural-search dependency. For instance, in this particular case it took k-NN team about a day to figure out and fix the problem, and during that time steps defined here wouldn't solve the issue.
I think we need following steps:
1. check if problem is coming from one of decencies of neural search plugin, e.g. ml-commons or k-NN
2. if that's the case then check if problem is fixed in that dependency repo
3. if 2 is true then continue with cleaning local gradle and maven caches <then steps you do have in PR>.

Reply:
Currently I don't know what other scenarios could occur. However I would be more than happy to edit further once Ive encountered those problems :) . I have edited the document to talk about this being a issue with a KNN dependency in that way we don't confuse the reader in thinking this is a solves all solution

Reply:
Okay, as we spoke offline I added a section under debugging that explains the major Dependencies that Neural-Search depends on. In that way it lets developers know how to think about dealing with build failures.